tests/smoke_test.go

```diff
@@ -0,0 +1,131 @@
+package tests
+
+import (
+	"errors"
+	"fmt"
+	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/apis/sparkoperator.k8s.io/v1beta2"
+	"github.com/mesosphere/kudo-spark-operator/tests/utils"
+	log "github.com/sirupsen/logrus"
+	v12 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestShuffleApp(t *testing.T) {
+	spark := utils.SparkOperatorInstallation{}
+	err := spark.InstallSparkOperator()
+	defer spark.CleanUp()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	const expectedExecutorCount = 2
+	const expectedGroupCount = 12000
+
+	jobName := "shuffle-app"
+	job := utils.SparkJob{
+		Name:     jobName,
+		Template: "spark-shuffle-job.yaml",
+		Params: map[string]interface{}{
+			"executor_count": expectedExecutorCount,
+			"group_count":    fmt.Sprintf("\"%d\"", expectedGroupCount),
+		},
+	}
+
+	err = spark.SubmitJob(&job)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = spark.WaitForJobState(job, v1beta2.RunningState, 5*time.Minute)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Wait for correct number of executors to show up
+	err = utils.Retry(30*time.Second, time.Second, func() error {
```

Comment:
can we extract timeouts either to a constant or to some structure with timeouts per test?

Reply:
done